tests/v2/actions/machinepools/machinepools.go

```diff
@@ -54,6 +58,17 @@ func MatchNodeRolesToMachinePool(nodeRoles NodeRoles, machinePools []apisV1.RKEM
 		return index, count
 	}
 
+	// If the nodeRole is for a Windows node, this separate check is needed. This is because
+	// the machinePoolConfig does not account for Windows nodes. This results in a scaling
+	// issue when working with Windows nodes.
```

Comment:
if you have time to add a comment justifying the double for loop, that would be nice to have here.